### PR TITLE
Seed refactoring

### PR DESCRIPTION
**Pull request by marcheing**

### CODE REVIEW

File: db/seeds.rb [hunk 1]
```diff
@@ -24,6 +24,13 @@
   { label: "Excellent", grade: 10, color: "00FF00", reading_group_id: reading_group.id }
 ])
 
+terrible = readings.first.id
+worrying = readings[2].id
+regular = readings[4].id
+satisfactory = readings[5].id
+good = readings[6].id
+excellent = readings.last.id
+
```
Comment (diegoamc) [inline db/seeds.rb]: This have only changed the location of the problem. Why don't we break the readings' and assign variables for each new created reading?

File: db/seeds.rb [hunk 2]
```diff
@@ -143,12 +150,12 @@
...
-  { beginning: 40, end: 60, comments: "", reading_id: readings[-5].id, metric_configuration_id: metric_configuration.id },
-  { beginning: 60, end: 100, comments: "", reading_id: readings[-7].id, metric_configuration_id: metric_configuration.id },
-  { beginning: 100, end: Float::INFINITY, comments: "", reading_id: readings[-9].id, metric_configuration_id: metric_configuration.id }
+  { beginning: 0, end: 10, comments: "", reading_id: excellent, metric_configuration_id: metric_configuration.id },
+  { beginning: 10, end: 20, comments: "", reading_id: good, metric_configuration_id: metric_configuration.id },
+  { beginning: 20, end: 40, comments: "", reading_id: satisfactory, metric_configuration_id: metric_configuration.id },
+  { beginning: 40, end: 60, comments: "", reading_id: regular, metric_configuration_id: metric_configuration.id },
+  { beginning: 60, end: 100, comments: "", reading_id: worrying, metric_configuration_id: metric_configuration.id },
+  { beginning: 100, end: Float::INFINITY, comments: "", reading_id: terrible, metric_configuration_id: metric_configuration.id }
 ])
```
Comment (diegoamc) [inline db/seeds.rb]: Can't we extract all those lines for creating instances to new methods?

**Comment by diegoamc** [comment]
Does it make sense to write unit tests for the seeds?

**Comment by rafamanzo** [comment]
Unit i don't think so. Integration maybe. Will be hard to test it.

Em qua, 29 de abr de 2015 às 21:34, Diego de Araújo Martinez Camarinha <
notifications@github.com> escreveu:

File: db/seeds.rb [hunk 3]
```diff
@@ -138,17 +168,19 @@
 
 ruby_configuration = KalibroConfiguration.create(name: "Ruby Configuration", description: "Example Ruby Configuration")
 
-metric_snapshot = NativeMetricSnapshot.create(name: "Pain", description: "Flog mede a tortuosidade do código-fonte. Quanto mais doloroso e difícil de testar, maior a pontuação, baseando-se na métrica ABC e boas práticas de Ruby. Extraído do blog do autor da métrica: http://jakescruggs.blogspot.com.br/2008/08/whats-good-flog-score.html", code: "flog", metric_collector_name: "MetricFu", scope: "METHOD")
+pain = NativeMetricSnapshot.create(name: "Pain",
+                                                              description: "Flog mede a tortuosidade do código-fonte. Quanto mais doloroso e difícil de testar, maior a pontuação, baseando-se na métrica ABC e boas práticas de Ruby. Extraído do blog do autor da métrica: http://jakescruggs.blogspot.com.br/2008/08/whats-good-flog-score.html",
+                                                              code: "flog", metric_collector_name: "MetricFu", scope: "METHOD")
```
Comment (danielkza) [inline db/seeds.rb]: I think something went wrong with the indentation here.

File: db/seeds.rb [hunk 4]
```diff
@@ -10,126 +10,156 @@
...
 
 ###############################################################################
 
-metric_snapshot = NativeMetricSnapshot.create(name: "Number of Public Attributes", description: "", code: "npa", metric_collector_name: "Analizo", scope: "CLASS")
+npa = NativeMetricSnapshot.create(name: "Number of Public Attributes", description: "",
+                                                              code: "npa", metric_collector_name: "Analizo", scope: "CLASS")
 
-metric_configuration = MetricConfiguration.create(metric_snapshot_id: metric_snapshot.id, weight: 1.0, aggregation_form: "AVERAGE", reading_group_id: reading_group.id, kalibro_configuration_id: kalibro_configuration.id)
+npa_metric_configuration = MetricConfiguration.create(metric_snapshot_id: npa.id, weight: 1.0, aggregation_form: "AVERAGE",
+                                                                                              reading_group_id: scholar.id, kalibro_configuration_id: first_configuration.id)
```
Comment (danielkza) [inline db/seeds.rb]: I think something went wrong with the indentation here.

File: db/seeds.rb [hunk 5]
```diff
@@ -10,126 +10,156 @@
...
 
 ###############################################################################
 
-metric_snapshot = NativeMetricSnapshot.create(name: "Number of Methods", description: "", code: "nom", metric_collector_name: "Analizo", scope: "CLASS")
+nom = NativeMetricSnapshot.create(name: "Number of Methods", description: "",
+                                                              code: "nom", metric_collector_name: "Analizo", scope: "CLASS")
 
-metric_configuration = MetricConfiguration.create(metric_snapshot_id: metric_snapshot.id, weight: 1.0, aggregation_form: "AVERAGE", reading_group_id: reading_group.id, kalibro_configuration_id: kalibro_configuration.id)
+nom_metric_configuration = MetricConfiguration.create(metric_snapshot_id: nom.id, weight: 1.0, aggregation_form: "AVERAGE",
+                                                                                                reading_group_id: scholar.id, kalibro_configuration_id: first_configuration.id)
```
Comment (danielkza) [inline db/seeds.rb]: I think something went wrong with the indentation here.

File: db/seeds.rb [hunk 6]
```diff
@@ -10,126 +10,156 @@
...
 
 ###############################################################################
 
-metric_snapshot = NativeMetricSnapshot.create(name: "Depth of Inheritance Tree", description: "", code: "dit", metric_collector_name: "Analizo", scope: "CLASS")
+dit = NativeMetricSnapshot.create(name: "Depth of Inheritance Tree", description: "",
+                                                            code: "dit", metric_collector_name: "Analizo", scope: "CLASS")
 
-metric_configuration = MetricConfiguration.create(metric_snapshot_id: metric_snapshot.id, weight: 1.0, aggregation_form: "AVERAGE", reading_group_id: reading_group.id, kalibro_configuration_id: kalibro_configuration.id)
+dit_metric_configuration = MetricConfiguration.create(metric_snapshot_id: dit.id, weight: 1.0, aggregation_form: "AVERAGE",
+                                                                                            reading_group_id: scholar.id, kalibro_configuration_id: first_configuration.id)
```
Comment (danielkza) [inline db/seeds.rb]: I think something went wrong with the indentation here.

File: db/seeds.rb [hunk 7]
```diff
@@ -10,126 +10,156 @@
...
+  { beginning: 5, end: Float::INFINITY,
+        comments: "Os métodos estão recebendo muitos parâmetros. Conjuntos de parâmetros parecidos sugerem a criação de uma classe contendo esses dados, e transferência dos métodos para essa classe.",
+        reading_id: worrying_reading.id, metric_configuration_id: anpm_metric_configuration.id }
 ])
 
 ###############################################################################
 
-metric_snapshot = NativeMetricSnapshot.create(name: "Depth of Inheritance Tree", description: "", code: "dit", metric_collector_name: "Analizo", scope: "CLASS")
+dit = NativeMetricSnapshot.create(name: "Depth of Inheritance Tree", description: "",
+                                                            code: "dit", metric_collector_name: "Analizo", scope: "CLASS")
```
Comment (danielkza) [inline db/seeds.rb]: I think something went wrong with the indentation here.

File: db/seeds.rb [hunk 8]
```diff
@@ -10,126 +10,156 @@
...
 
 ###############################################################################
 
-metric_snapshot = NativeMetricSnapshot.create(name: "Average Number of Parameters per Method", description: "", code: "anpm", metric_collector_name: "Analizo", scope: "CLASS")
+anpm = NativeMetricSnapshot.create(name: "Average Number of Parameters per Method", description: "",
+                                                                code: "anpm", metric_collector_name: "Analizo", scope: "CLASS")
 
-metric_configuration = MetricConfiguration.create(metric_snapshot_id: metric_snapshot.id, weight: 1.0, aggregation_form: "AVERAGE", reading_group_id: reading_group.id, kalibro_configuration_id: kalibro_configuration.id)
+anpm_metric_configuration = MetricConfiguration.create(metric_snapshot_id: anpm.id, weight: 1.0, aggregation_form: "AVERAGE",
+                                                                                                  reading_group_id: scholar.id, kalibro_configuration_id: first_configuration.id)
```
Comment (danielkza) [inline db/seeds.rb]: I think something went wrong with the indentation here.

File: db/seeds.rb [hunk 9]
```diff
@@ -10,126 +10,156 @@
...
+  { beginning: 37, end: Float::INFINITY,
+        comments: "Os métodos estão muito longos. Tente quebrá-los em métodos menores, cada um no seu nível de abstração. Forte candidato a novo método é o código em maior aninhamento.",
+        reading_id: worrying_reading.id, metric_configuration_id: amloc_metric_configuration.id }
 ])
 
 ###############################################################################
 
-metric_snapshot = NativeMetricSnapshot.create(name: "Average Number of Parameters per Method", description: "", code: "anpm", metric_collector_name: "Analizo", scope: "CLASS")
+anpm = NativeMetricSnapshot.create(name: "Average Number of Parameters per Method", description: "",
+                                                                code: "anpm", metric_collector_name: "Analizo", scope: "CLASS")
```
Comment (danielkza) [inline db/seeds.rb]: I think something went wrong with the indentation here.

File: db/seeds.rb [hunk 10]
```diff
@@ -10,126 +10,156 @@
...
 
 ###############################################################################
 
-metric_snapshot = NativeMetricSnapshot.create(name: "Average Method Lines of Code", description: "", code: "amloc", metric_collector_name: "Analizo", scope: "CLASS")
+amloc = NativeMetricSnapshot.create(name: "Average Method Lines of Code", description: "",
+                                                                code: "amloc", metric_collector_name: "Analizo", scope: "CLASS")
 
-metric_configuration = MetricConfiguration.create(metric_snapshot_id: metric_snapshot.id, weight: 1.0, aggregation_form: "AVERAGE", reading_group_id: reading_group.id, kalibro_configuration_id: kalibro_configuration.id)
+amloc_metric_configuration = MetricConfiguration.create(metric_snapshot_id: amloc.id, weight: 1.0, aggregation_form: "AVERAGE",
+                                                                                                  reading_group_id: scholar.id, kalibro_configuration_id: first_configuration.id)
```
Comment (danielkza) [inline db/seeds.rb]: I think something went wrong with the indentation here.

File: db/seeds.rb [hunk 11]
```diff
@@ -138,17 +201,22 @@
...
-  { beginning: 10, end: 20, comments: "", reading_id: readings[-3].id, metric_configuration_id: metric_configuration.id },
-  { beginning: 20, end: 40, comments: "", reading_id: readings[-4].id, metric_configuration_id: metric_configuration.id },
-  { beginning: 40, end: 60, comments: "", reading_id: readings[-5].id, metric_configuration_id: metric_configuration.id },
-  { beginning: 60, end: 100, comments: "", reading_id: readings[-7].id, metric_configuration_id: metric_configuration.id },
-  { beginning: 100, end: Float::INFINITY, comments: "", reading_id: readings[-9].id, metric_configuration_id: metric_configuration.id }
-])
+pain = NativeMetricSnapshot.create(
+  name: "Pain", code: "flog", metric_collector_name: "MetricFu", scope: "METHOD",
+  description: "Flog mede a tortuosidade do código-fonte. Quanto mais doloroso e difícil de testar, maior a pontuação, baseando-se na métrica ABC e boas práticas de Ruby. Extraído do blog do autor da métrica: http://jakescruggs.blogspot.com.br/2008/08/whats-good-flog-score.html"
+)
```
Comment (pedroscocco) [inline db/seeds.rb]: All other variables have the name os the metric's code, not the metric's name

**Pull request merged**